source/Applications/Advanced/Downsample.m

```diff
@@ -11,13 +11,24 @@
 
 dataFile = [char(System.Environment.GetFolderPath(System.('Environment+SpecialFolder.CommonApplicationData'))),'/Zivid/Zivid3D.zdf'];
 disp(['Reading ZDF frame from file: ',dataFile]);
-[xyz,rgb,snr] = readZDF(dataFile);
+frame = Zivid.NET.Frame(dataFile);
+
+pointCloudZivid = frame.PointCloud();
+xyz = single(frame.PointCloud.CopyPointsXYZ);
+rgba = uint8(frame.PointCloud.CopyColorsRGBA);
+rgb = rgba(:,:,1:3);
 
 pointCloudOriginal = pointCloud(xyz,'color',rgb);
 
-downsamplingFactor = 4;
-disp(['Downsampling point cloud with factor ',num2str(downsamplingFactor)]);
-[xyzDownsampled,rgbDownsampled] = downsample(xyz,rgb,snr,downsamplingFactor);
+disp(['Size of point cloud before downsampling: ', num2str(pointCloudZivid.Size), ' data points']);
+
+pointCloudZivid.Downsample(Zivid.NET.('PointCloud+Downsampling.By2x2'));
+
+disp(['Size of point cloud before downsampling: ', num2str(pointCloudZivid.Size), ' data points']);
```

Comment:
@robertkt1 you already have this on line 23

Reply:
Ah forgot to edit to "size of point cloud after downsampling". To compare the sizes before and after. 
Think it is a nice to have?